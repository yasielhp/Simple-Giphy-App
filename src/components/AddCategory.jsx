import { useState } from "react"
import { PropTypes } from 'prop-types';
import { Input } from "@nextui-org/react";
export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {setInputValue(e.target.value)}
  const handleSubmit = (e) => {
    e.preventDefault()
    if(inputValue.trim().length <= 1) return
    onNewCategory(inputValue.trim())
    setInputValue('')
  }
  return (
    <form onSubmit={handleSubmit}>
      <Input
        fullWidth
        clearable
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        label="Search for your gif"
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}