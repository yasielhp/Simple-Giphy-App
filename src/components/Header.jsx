import { Text, Grid, Switch, useTheme, Link } from '@nextui-org/react'
import { useTheme as useNextTheme } from 'next-themes'


export const Header = () => {
  const { setTheme } = useNextTheme();
  const { isDark } = useTheme();
  const handleThemeChange = (e) => setTheme(e.target.checked ? 'dark' : 'light')
  return (
    <Grid.Container justify='space-between' alignItems='center' css={{px:5}}>
      <Grid css={{ display: 'flex', alignItems: 'baseline' }} >
        <Link href='/'>
          <Text span size={48}css={{ textGradient: "-85deg, $blue600 -40%, $green600 120%" }}>GIF</Text>
          <Text h1 css={{ color: "$blue600" }}>SEA</Text>
        </Link>
      </Grid>
      <Switch color="success" checked={isDark} onChange={handleThemeChange}  />
    </Grid.Container>
  )
}
