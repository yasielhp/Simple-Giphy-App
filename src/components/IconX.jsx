export const IconX = ({
  fill = 'currentColor',
  filled,
  size,
  height,
  width,
  label,
  ...props
}) => {
    return (
      <svg
        width={size || width || 24}
        height={size || height || 24}
        viewBox="0 0 24 24"
        stroke="currentColor"
        strokeWidth={3}
        {...props}
      >
        <path
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M6 18L18 6M6 6l12 12"
      />
      </svg>
    );
};



