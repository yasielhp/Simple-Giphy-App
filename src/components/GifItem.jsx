import { Grid, Text, Col, Row, Button, Card, Modal, Image } from '@nextui-org/react';
import { useState } from 'react';
import PropTypes from 'prop-types';
import { IconX } from './IconX';
import { saveAs } from 'file-saver'
export const GifItem = ({ title, url }) => {
  const [visible, setVisible] = useState(false)
  const handler = () => setVisible(true)
  const closeHandler = () => { setVisible(false) }
  const downloadHandler = () => {
    saveAs(url, title)
  }

  return (
    <>
      <Grid xs={15} sm={4} md={3} justify='center'>
        <Card css={{ w: "400px", h: "400px" }} isPressable isHoverable onPress={handler}>
          <Card.Body css={{ p: 0 }} >
            <Card.Image src={url} width="100%" height="100%" objectFit='cover' alt={title}/>
          </Card.Body>
          <Card.Footer isBlurred css={{ position: "absolute", bgBlur: "#00000066", borderTop: "$borderWeights$light solid rgba(255, 255, 255, 0.2)", bottom: 0, zIndex: 1,}}>
            <Row css={{alignItems:"center"}}>
              <Col>
                <Text color="#fff">{title}</Text>
              </Col>
              <Col >
                <Row justify="flex-end" >
                  <Button flat auto rounded color="success" onPress={downloadHandler} >
                    <Text css={{ color: "inherit" }} size={12} weight="bold" transform="uppercase">
                      Download
                    </Text>
                  </Button>
                </Row>
              </Col>
            </Row>
          </Card.Footer>
        </Card>
      </Grid>
      <Modal blur noPadding aria-labelledby="modal-title" open={visible} onClose={closeHandler}>
        <Modal.Header css={{ position: "absolute", zIndex: 1, top: 5, right: 5, gap: 5, justifyContent: 'space-between' }}>
          <Button auto onPress={closeHandler} rounded color="error" css={{ p: 8 }} >
            <IconX fill="#fff" />
          </Button>
        </Modal.Header>
        <Modal.Body css={{ p: 0 }}>
          <Image src={url} width="100%" height="100%" objectFit='cover' alt={title}/>
        </Modal.Body>
        <Modal.Footer css={ {p:6 }}>
        <Row gap={1} css={{display:'flex' ,justifyContent:'space-between'}}>
            <Col css={{textAlign:'left'}}>
                <Text size={14}>{title}</Text>
            </Col>
            <Col css={{alignContent:'center', display:'flex', justifyContent:"end"}}>
              <Button flat auto rounded color="success" onPress={downloadHandler}>
                <Text css={{ color: "inherit" }} size={12} weight="bold" transform="uppercase">
                  Download
                </Text>
              </Button>
            </Col>
          </Row>

        </Modal.Footer>
      </Modal>
    </>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
