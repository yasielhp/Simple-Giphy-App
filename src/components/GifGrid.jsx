import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { PropTypes } from 'prop-types';
import { Grid, Text, Spacer, Loading } from '@nextui-org/react';

export const GifGrid = ({ category }) => {
  const { images, isLoading} = useFetchGifs(category);


  return (
    <>
      <Spacer y={1}/>
      <Text h2 size={20}  weight="bold">Search results for:</Text>
      <Text css={{ color: '$green600'}}> {category}</Text>
      <Spacer y={1}/>
      {
        isLoading && <Grid css={{justifyContent:'center', display:'flex'}}>
          <Loading size="lg" color="success" type="points" />
        </Grid>
      }
      <Grid.Container gap={2} justify="center">
        {
          images.map((image) => (
            <GifItem key={image.id} {...image} />
          ))
        }
      </Grid.Container>
      <Spacer y={1}/>
    </>
  )
}

GifGrid.protoType = {
  category: PropTypes.string.isRequired
}