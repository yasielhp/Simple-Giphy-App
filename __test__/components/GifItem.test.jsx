import { render, screen } from "@testing-library/react"
import { GifItem } from '../../src/components/GifItem';

describe('Test in <GifItem/>', () => {
  const title = 'Test title'
  const url = 'https://via.placeholder.com/150'

  test('- Debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('- Debe de tener una imagen con la URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('- Debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})