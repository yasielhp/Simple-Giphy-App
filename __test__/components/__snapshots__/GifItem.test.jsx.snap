// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> - Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-icdMRcE-css nextui-grid-item xs sm md"
  >
    <div
      class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-PJLV nextui-c-BDLTQ-jyIjWj-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-BDLTQ-igmQosi-css"
      role="button"
      tabindex="0"
    >
      <div
        class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
      >
        <div
          class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ihinyfY-css nextui-image-container nextui-image--loading"
          data-state="loading"
        >
          <div
            class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
          />
          <img
            alt="Test title"
            class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
            data-state="loading"
            height="100%"
            src="https://via.placeholder.com/150"
            width="100%"
          />
        </div>
      </div>
      <div
        class="nextui-c-cakKrd nextui-c-cakKrd-gjCMAI-isBlurred-true nextui-c-cakKrd-ienEzvy-css"
      >
        <div
          class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ilailYI-css"
        >
          <div
            class="nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css"
          >
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igYRsiq-css"
            >
              Test title
            </p>
          </div>
          <div
            class="nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css"
          >
            <div
              class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-iipPaye-css"
            >
              <button
                class="nextui-c-iWjDFM nextui-c-PJLV nextui-c-ldmKBv nextui-c-iWjDFM-hfnhFm-color-success nextui-c-iWjDFM-gkIlwi-size-md nextui-c-iWjDFM-cwXrJp-borderWeight-normal nextui-c-iWjDFM-cyFBRX-flat-true nextui-c-iWjDFM-exaRme-auto-true nextui-c-iWjDFM-hzYxzX-rounded-true nextui-c-iWjDFM-gglxka-cv nextui-c-iWjDFM-gmvfdi-cv nextui-c-iWjDFM-euJLgm-cv nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-iWjDFM-iPJLV-css nextui-button nextui-button--ready"
                data-state="ready"
                type="button"
              >
                <span
                  class="nextui-button-text"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-hyvuql-weight-bold nextui-c-PJLV-ikBSpPJ-css"
                  >
                    Download
                  </p>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
